docs/platform_developers/create_server_extension.rst

```diff
@@ -0,0 +1,127 @@
+*******************************
+Creating a new server extension
+*******************************
+
+An Inmanta server extension adds extra functionality to the Inmanta core. Each extension consists of one or more server slices
```

Comment:
We should add a sentence that explains slices as well and how they relate to extensions. The reasoning is:
- Slices are components in the service orchestrator. A slice can be responsible for API endpoints or provide internal services to other slices.
- Extensions are separate python packages with their own release cycle that can add additional slices (components) to the orchestrator. All slices of the core service orchestrator are part of the core server extension.